papis/commands/update.py

```diff
@@ -39,25 +120,223 @@
+                    document[key].append(papis.utils.clean_document_name(file))
+                else:
+                    document[key].append(value)
+        else:
+            document[key] = value
+
+
+def run_append(
+    document: papis.document.DocumentLike,
+    to_append: List[Tuple[str, str]],
+    key_types: Dict[str, type],
+) -> bool:
+    """
+    Processes a list of `to_append` tuples and applies the resulting changes
+    to the input document. Each tuple is (KEY, VALUE) and results in appending
+    the VALUE to the KEY item.
+
+    :returns: A boolean indicating whether the update was successful.
+    """
+    success = True
+    processed_lists = set()
+    for key, value in to_append:
+        if key in set().union(document, key_types):
+            value = papis.format.format(value, document, default=value)
+            # TODO: check if below is right
+            type_doc = type(document.get(key))
+            type_conf = key_types.get(key)
+            if type_doc is str or (type_doc is type(None) and type_conf is str):
+                document[key] = document.setdefault(key, "") + value
+                document[key] += value
+            elif type_doc is list or (type_doc is type(None) and type_conf is list):
+                processed_lists.add(key)
+                value = try_parsing_str(key, value)
+                # NOTE: while in `papis update --set`, i handled both notes and
+                #       files, here it's just files. if you append to notes, that
+                #       means you're appending a string to a string, and if u do
+                #       that with a filename, you're responsible for your own
+                #       safety ;).
+                if key == "files" and isinstance(value, str):
+                    value = papis.utils.clean_document_name(value)
+                document[key] = document.setdefault(key, [])
+                document[key].append(value)
+            else:
+                logger.error(
+                    "Items of key '%s' have the type '%s', for which Papis "
+                    "doesn't support the append operation.",
+                    key,
+                    type(document.get(key)).__name__ or key_types.get(key),
+                )
+                success = False
+                break
+        else:
+            logger.error(
+                "We cannot append to key '%s', because we do not know the "
+                "intended type. Please use `papis update --set` instead.",
+                key,
+            )
+            success = False
+            break
+
+    for key in processed_lists:
+        try:
+            document[key] = list(set(document[key]))
+        except Exception:
+            logger.warn(
+                "You're appending a `list` or `dict` item, which are not "
+                "hashable. Papis therefore couldn't check whether you've "
+                "introduced a duplicate in the list."
+            )
```

Comment:
This seems like a slightly related "fix" to add here.. maybe it should be a `papis doctor` check instead?

Reply:
Yeah, I meant checking `processed_lists` and issuing this warning.